Mathlib/Algebra/Order/Ring/Ordering/Basic.lean

```diff
@@ -0,0 +1,189 @@
+/-
+Copyright (c) 2024 Florent Schaffhauser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Florent Schaffhauser, Artie Khovanov
+-/
+import Mathlib.Algebra.Field.IsField
+import Mathlib.Algebra.Order.Ring.Ordering.Defs
+import Mathlib.Algebra.Ring.SumsOfSquares
+import Mathlib.Tactic.FieldSimp
+import Mathlib.Tactic.LinearCombination
+
+/-!
+# Ring orderings
+
+We prove basic properties of (pre)orderings on rings and their supports.
+
+## References
+
+- [*An introduction to real algebra*, T.Y. Lam][lam_1984]
+
+-/
+
+variable {R : Type*} [CommRing R] {P : RingPreordering R}
+
+/-!
+### Preorderings
+-/
+
+namespace RingPreordering
+
+@[mono]
+theorem toSubsemiring_strictMono : StrictMono (toSubsemiring : RingPreordering R → _) :=
+  fun _ _ => id
+
+theorem toSubsemiring_lt_toSubsemiring {P₁ P₂ : RingPreordering R} :
```

Comment:
I'd personally put this just above the `le` version.

Reply:
I'd put it just below, but yes